### PR TITLE
fix: revert #144

### PR DESCRIPTION
**Pull request by yzh119**
#144 multiplies q by sm_scale, which may cause numerical issues when q is a fp8 data type.
This PR reverts the change and multiply logits by sm_scale at the end of aggregation.

### CODE REVIEW

**Comment by yzh119** [comment]
Okay I think I made a mistake here, in decode kernels, we first cast q to fp16 then multiply it by sm_scale, so it's not a issue.
